### ISSUE TITLE
pebble `make-dirs` permission is masked by pebble daemon's umask

### ISSUE THREAD

**Issue by weiiwang01** [canonical/pebble#372]
When running the Pebble `make-dirs` command, the resulting directory permissions may not match the requested settings, as they are affected by the umask of the pebble daemon.

To reproduce:

```bash
( umask 077; ./pebble run )

./pebble mkdir -m=777 /tmp/test-mkdir
ls -lah /tmp/test-mkdir/.
total 8.0K
drwx------  2 weii-wang weii-wang 4.0K Feb 28 05:02 . # it should be 777 instead of 700
drwxrwxrwt 16 root      root      4.0K Feb 28 05:02 ..
```

**Comment by weiiwang01** [comment]
Well the simplest way to fix this issue is to just call `syscall.Umask(0)` when starting the daemon or fix the issue in `internals/osutil/mkdirallchown.go`, which seems not trivial. I wonder which approach would be better.

**Comment by weiiwang01** [comment]
Oh, I forgot that the umask is inherited by child processes, so it's probably not a good idea to change the umask of Pebble itself.

**Comment by benhoyt** [comment]
It looks like this might be a duplicate of https://github.com/canonical/pebble/issues/80 -- however, when that issue was fixed (here: https://github.com/canonical/pebble/commit/a2161d464bee0e7532e7ef13d52e677fb4ab28ae) it looks like it was only fixed for write-file and not make-dirs, so keeping this open to track that.

**Comment by IronCore864** [comment]
This issue was accidentally closed by https://github.com/canonical/pebble/pull/405. Reopen, and will be closed by #423.
